ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/Offset.scala

```diff
@@ -21,22 +20,18 @@ import com.google.protobuf.ByteString
   * less than newer offsets.
   *
   */
-final class Offset(private val value: ByteString) extends AnyVal with Ordered[Offset] {
+sealed trait OffsetTag
 
-  override def compare(that: Offset): Int =
-    Offset.comparator.compare(value, that.value)
+object Offset {
 
-  def toByteArray: Array[Byte] = value.toByteArray
+  val tag = Tag.of[OffsetTag]
 
-  def toInputStream: InputStream = value.newInput()
-}
+  def apply(s: Bytes): Offset = tag(s)
 
-object Offset {
-  private val comparator = ByteString.unsignedLexicographicalComparator()
+  def unwrap(x: Offset): Bytes = tag.unwrap(x)
```

Comment:
I'd like to keep the `to*` methods delegating to `unwrap` for readability. A `toByteBuffer` and a `toHexString` might make sense too. Calling `unwrap` directly seems like a bad sign to me.

Reply:
I put back to `to*` and the `from*`. 
I added only `toHexString` as `toByteBuffer` is not used.